Jellyfin.Api/Helpers/ItemHelper.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Jellyfin.Api.Models.UserDtos;
+using LrcParser.Model;
+using LrcParser.Parser;
+using MediaBrowser.Controller.Entities;
+
+namespace Jellyfin.Api.Helpers
+{
+    /// <summary>
+    /// Item helper.
+    /// </summary>
+    public static class ItemHelper
+    {
+        /// <summary>
+        /// Opens lyrics file, converts to a List of Lyrics, and returns it.
+        /// </summary>
+        /// <param name="item">Requested Item.</param>
+        /// <returns>Collection of Lyrics.</returns>
+        internal static object? GetLyricData(BaseItem item)
+        {
+            // Find all classes that implement ILyricsProvider Interface
+            var foundLyricProviders = System.Reflection.Assembly.GetExecutingAssembly()
+                .GetTypes()
+                .Where(type => typeof(ILyricsProvider).IsAssignableFrom(type) && !type.IsInterface);
+
+            if (!foundLyricProviders.Any())
+            {
+                return null;
+            }
+
+            foreach (var provider in foundLyricProviders)
+            {
+                ILyricsProvider? newProvider = Activator.CreateInstance(provider) as ILyricsProvider;
+                if (newProvider is not null)
+                {
+                    newProvider.Process(item);
+                    if (newProvider.HasData)
```

Comment:
Seems pointless to keep state in the provider. Why not just return the result (or null) directly?